test/pipelinefinally_test.go

```diff
@@ -269,7 +318,17 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_InvalidTaskResult_Failure(t *test
 					t.Errorf("Error resolving task result reference in a finally task %s", n)
 				}
 			}
-		case n == "finaltaskconsumingdagtask1" || n == "finaltaskconsumingdagtask4":
+		case n == "finaltaskconsumingdagtask5inwhenexpressionstrue":
+			if err := WaitForTaskRunState(ctx, c, taskrunItem.Name, TaskRunSucceed(taskrunItem.Name), "TaskRunSuccess"); err != nil {
```

Comment:
do we want to be waiting here? I think we already waited for the pipelinerun as a whole to get into the state we expected (or maybe im wrong?) - if the taskruns aren't in the expected state after the pipelinerun has completed, i think something has gone wrong?

Reply:
that's right, replaced with checking the taskrun status